src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -236,7 +298,14 @@ protected override void Dispose(bool disposing)
             finally
             {
                 _stream = null;
-                _buffer = null;
+
+                // Don't set the buffer to null, to avoid a NullReferenceException
+                // when users have a race condition in their code (i.e. they call
+                // Close when calling another method on Stream like Read).
```

Comment:
Is this related to the FileStream rewrite?  Seems like the answer is "no", in which case let's not do it in this PR.

Reply:
It is, this is what `FileStream` did so far (5.0 branch):

https://github.com/dotnet/runtime/blob/832a65623d180412fcbaefd68aa57065341db931/src/libraries/System.Private.CoreLib/src/System/IO/FileStream.Windows.cs#L258-L260

If I remember correctly if I remove this, there is one test that will from time to time fail with `NullReferenceException`.